### PR TITLE
Finish daos and tests class

### PR DESCRIPTION
**Pull request by shaharshw**
without the new api

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    private val ownerMapper = RecordMapper<Record, Owner>
+    { record ->
+        Owner(
+            id = record[OwnerTable.instance.id],
+            name = record[OwnerTable.instance.name],
+            companyId = record[OwnerTable.instance.companyId],
+            employeeId = record[OwnerTable.instance.employeeId].toString(),
+            firstName = null,
+            lastName = null
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt]: If you are using firstName and lastName then you get them from the name itself and not put null.

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        )
+    }
+
+    fun getAllOwners(): List<Owner> {
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt]: By companyId.
We should fetch by the companyId provided.
The is true to all the dao functions.

File: src/main/kotlin/com/hibob/academy/dao/PetDao.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        return sql.insertInto(p)
+            .set(p.id, pet.id)
+            .set(p.name, pet.name)
+            .set(p.type, pet.type.name)
+            .set(p.dateOfArrival, pet.dataOfArrival)
+            .set(p.companyId, pet.companyId)
+            .execute()
+    }
+
+    fun getAllPets(): List<Pet> {
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetDao.kt]: Same here regarding the companyId.

File: src/main/kotlin/com/hibob/academy/dao/PetDao.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        return sql.select(p.id, p.name, p.type, p.dateOfArrival, p.companyId)
+            .from(p)
+            .fetch (petMapper)
+    }
+
+    fun getPetsByType(petType: PetType): List<PetWithoutType> {
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetDao.kt]: Same here regarding the companyId

File: src/main/kotlin/com/hibob/academy/entity/types.kt [hunk 5]
```diff
@@ -1,20 +1,41 @@
 package com.hibob.academy.entity
 
-import java.sql.Timestamp
+import java.sql.Date
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/entity/types.kt]: Why not use LocalDate?

File: src/main/kotlin/com/hibob/academy/entity/types.kt [hunk 6]
```diff
@@ -1,20 +1,41 @@
...
+        }
+    }
+}
+
+data class PetWithoutType(
+    val id: Long,
+    val name: String,
+    val dataOfArrival : Date,
+    val companyId : Long
+)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/entity/types.kt]: Why do we need this data class? All pets should have type otherwise they are not pets🤷🏻‍♂️

File: src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql [hunk 7]
```diff
@@ -1,11 +1,10 @@
 create table pets
 (
-    id              UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    id              SERIAL PRIMARY KEY,
```
Comment (omermelamed) [inline src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql]: Should be in a different pr.

File: src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql [hunk 8]
```diff
@@ -1,12 +1,12 @@
 create table owner
 (
-    id          UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    id          SERIAL PRIMARY KEY,
```
Comment (omermelamed) [inline src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql]: Same, should be in a different pr.

File: src/main/resources/db/migration/V202409111725__add_unique_index_idx_owner.sql [hunk 9]
```diff
@@ -0,0 +1,4 @@
+DROP INDEX IF EXISTS idx_owner_company_id_employee_id;
```
Comment (omermelamed) [inline src/main/resources/db/migration/V202409111725__add_unique_index_idx_owner.sql]: All migrations should be in a different prs.

File: src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+            employeeId = "123",
+            firstName = "Shahar",
+            lastName = "Shwartz"
+        )
+
+        ownerDao.createOwner(owner1)
+        ownerDao.createOwner(owner2)
+
+        val owners = ownerDao.getAllOwners()
+        assertEquals(1, owners.size)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt]: Why do you check the size and not the data that returned?

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+            name = "Buddy",
+            type = PetType.CAT,
+            dataOfArrival = Date(2021, 1, 1),
+            companyId = companyId
+        )
+
+        petDao.createPet(pet1)
+        val pets = petDao.getPetsByType(PetType.DOG)
+
+        assertEquals(0, pets.size)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt]: Why do we check the size and not the data that returned?

**Comment by omermelamed** [comment]
Lets add more tests that cover all scenarios.

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt (hunk 1 above)
Comment (shaharshw) [reply src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt]: ok, fixed it

File: src/main/kotlin/com/hibob/academy/entity/types.kt (hunk 5 above)
Comment (shaharshw) [reply src/main/kotlin/com/hibob/academy/entity/types.kt]: change it to LocalDate

File: src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+        )
+
+        ownerDao.createOwner(owner)
+        val owners = ownerDao.getAllOwnersByCompanyId(companyId)
+
+        assertEquals(1, owners.size)
+        assertEquals("Shahar Shwartz", owners[0].name)
+        assertEquals("123", owners[0].employeeId)
+        assertEquals("Shahar", owners[0].firstName)
+        assertEquals("Shwartz", owners[0].lastName)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt]: Instead of all these asserts lets compare lists and assert once.

File: src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+        ownerDao.createOwner(owner1)
+        ownerDao.createOwner(owner2)
+
+        val owners = ownerDao.getAllOwnersByCompanyId(companyId)
+
+        assertEquals(1, owners.size)
+        assertEquals("Shahar Shwartz", owners[0].name)
+        assertEquals("123", owners[0].employeeId)
+        assertEquals("Shahar", owners[0].firstName)
+        assertEquals("Shwartz", owners[0].lastName)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt]: Same comment as the test above.

File: src/main/kotlin/com/hibob/academy/entity/types.kt [hunk 14]
```diff
@@ -1,20 +1,35 @@
 package com.hibob.academy.entity
 
-import java.sql.Timestamp
+import java.time.LocalDate
+import java.util.*
 
 data class Pet(
-    val id: Long,
+    val id: UUID,
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/entity/types.kt]: Long

File: src/main/kotlin/com/hibob/academy/entity/types.kt [hunk 15]
```diff
@@ -1,20 +1,35 @@
...
-    val type : String,
-    val dataOfArrival : Timestamp,
+    val type : PetType,
+    val dataOfArrival : LocalDate,
     val companyId : Long
 )
 
 data class Owner(
-    val id: Long,
+    val id: UUID,
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/entity/types.kt]: Long

File: src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt [hunk 16]
```diff
@@ -15,8 +16,8 @@ class OwnerResource {
     @GET
     fun getALLOwners() : Response {
         return Response.ok(listOf(
-            Owner(1, "Shahar", 1, 1, null, null),
-            Owner(2, "Alex", 1, 2, null, null)
+            Owner(UUID.randomUUID(), "Shahar", 1L, "1", null, null),
+            Owner(UUID.randomUUID(), "Alex", 2L, "2", null, null)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt]: Long

File: src/main/kotlin/com/hibob/academy/resource/PetResource.kt [hunk 17]
```diff
@@ -15,8 +19,8 @@ class PetResource {
     @GET
     fun getALLPets() : Response{
         val pets = listOf(
-            Pet(1, "Bobby", "Dog", java.sql.Timestamp(System.currentTimeMillis()), 1),
-            Pet(2, "Kitty", "Cat", java.sql.Timestamp(System.currentTimeMillis()), 1)
+            Pet(UUID.randomUUID(), "Bobby", PetType.DOG, LocalDate.parse("2021-01-01"), 1),
+            Pet(UUID.randomUUID(), "Kitty", PetType.CAT, LocalDate.parse("2021-01-01"), 1)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/resource/PetResource.kt]: Long

File: src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+import org.springframework.beans.factory.annotation.Autowired
+import java.util.*
+
+@BobDbTest
+class OwnerDaoTest @Autowired constructor(private val sql: DSLContext) {
+
+    private val ownerDao = OwnerDao(sql)
+    val table = OwnerTable.instance
+    val companyId = 1L
+    val id = UUID.randomUUID()
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt]: Long

**Review by omermelamed — APPROVED** [review]
